### PR TITLE
Fix broken code coverage

### PR DESCRIPTION
**Pull request by moon0326**
### All Submissions:

* [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

1. Increased PHP Code Coverage workflow timeout to 20 mins.
2. Excluded `WC_Admin_Tests_Reports_Orders_Stats` test from the code coverage -- this test takes too long (6k lines of code 😮 ). We should refactor the test.
3. Added clean-up code to avoid changing global state.

### How to test the changes in this Pull Request:

1. Make sure `Run code coverage on PR / Code coverage (PHP 7.4, WP Latest) (pull_request) ` passes in this PR.
2. Optionally, you can run `pnpm nx test-code-coverage woocommerce` in your local.

### Changelog entry
N/A

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/api/product-attributes.php [hunk 1]
```diff
@@ -30,6 +30,14 @@ public function setUp(): void {
 		);
 	}
 
+	public static function tearDownAfterClass(): void {
+		parent::tearDownAfterClass();
+		global $wpdb;
+		parent::tearDownAfterClass(); //
```
Comment (chihsuan) [inline plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/api/product-attributes.php]: Why do we need to call `parent::tearDownAfterClass()` twice in this function? 👀

File: plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/api/reports-categories.php [hunk 2]
```diff
@@ -187,7 +187,6 @@ public function test_get_reports_categories_sort() {
 		$order->save();
 
 		WC_Helper_Queue::run_all_pending();
-		CategoryLookup::instance()->regenerate();
```
Comment (chihsuan) [inline plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/api/reports-categories.php:190]: Is it intended to remove this line?

File: plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/reports/class-wc-tests-reports-orders-stats.php [hunk 3]
```diff
@@ -11,6 +11,8 @@
 
 /**
  * Class WC_Admin_Tests_Reports_Orders_Stats
+ * Excluding it from code coverage until we refactor this test.
+ * @group run-in-seprate-process
```
Comment (ilyasfoo) [inline plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/reports/class-wc-tests-reports-orders-stats.php]: Perhaps typo here, `run-in-separate-process`?

**Comment by chihsuan** [comment]
Nice work! `Run code coverage on PR / Code coverage (PHP 7.4, WP Latest) (pull_request)` does passes.

But I was not able to run it in my local. Do we need to run other commands before running this?

```
❯ pnpm nx test-code-coverage woocommerce

> nx run woocommerce:test-code-coverage
[PHP Warning:  fopen(/var/folders/1h/svm7fpx53nj1rddp8fxywghh0000gn/T/wordpress-tests-lib/includes/functions.php): failed to open stream: No such file or directory in /Users/chihsuan/Projects/woocommerce/plugins/woocommerce/tests/Tools/CodeHacking/CodeHacker.php on line 462]

Warning: fopen(/var/folders/1h/svm7fpx53nj1rddp8fxywghh0000gn/T/wordpress-tests-lib/includes/functions.php): failed to open stream: No such file or directory in /Users/chihsuan/Projects/woocommerce/plugins/woocommerce/tests/Tools/CodeHacking/CodeHacker.php on line 462
[PHP Warning:  require_once(/var/folders/1h/svm7fpx53nj1rddp8fxywghh0000gn/T/wordpress-tests-lib/includes/functions.php): failed to open stream: "Automattic\WooCommerce\Testing\Tools\CodeHacking\CodeHacker::stream_open" call failed in /Users/chihsuan/Projects/woocommerce/plugins/woocommerce/tests/legacy/bootstrap.php on line 63]

Warning: require_once(/var/folders/1h/svm7fpx53nj1rddp8fxywghh0000gn/T/wordpress-tests-lib/includes/functions.php): failed to open stream: "Automattic\WooCommerce\Testing\Tools\CodeHacking\CodeHacker::stream_open" call failed in /Users/chihsuan/Projects/woocommerce/plugins/woocommerce/tests/legacy/bootstrap.php on line 63
[PHP Fatal error:  require_once(): Failed opening required '/var/folders/1h/svm7fpx53nj1rddp8fxywghh0000gn/T/wordpress-tests-lib/includes/functions.php' (include_path='.:/opt/homebrew/Cellar/php@7.4/7.4.28_1/share/php@7.4/pear') in /Users/chihsuan/Projects/woocommerce/plugins/woocommerce/tests/legacy/bootstrap.php on line 63]

Fatal error: require_once(): Failed opening required '/var/folders/1h/svm7fpx53nj1rddp8fxywghh0000gn/T/wordpress-tests-lib/includes/functions.php' (include_path='.:/opt/homebrew/Cellar/php@7.4/7.4.28_1/share/php@7.4/pear') in /Users/chihsuan/Projects/woocommerce/plugins/woocommerce/tests/legacy/bootstrap.php on line 63
ERROR: Something went wrong in @nrwl/run-commands - Command failed: RUN_CODE_COVERAGE=1 bash tests/bin/phpunit.sh

———————————————————————————————————————————————

>  NX   ERROR  Running target "woocommerce:test-code-coverage" failed

  Failed tasks:

  - woocommerce:test-code-coverage

  Hint: run the command with --verbose for more details.
```

**Comment by ObliviousHarmony** [comment]
@chihsuan This is because you need to run the test installation script in `plugins/woocommerce/tests/bin/install.sh`. The test setup right now is pretty environment specific. I'm currently changing it to run in `wp-env`, that way the setup is very easy and convenient. For this Pr though, just running that should suffice.

File: plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/api/product-attributes.php (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/api/product-attributes.php]: That's an error :) Thank you for catching it. I'll update.

File: plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/api/reports-categories.php (hunk 2 above)
Comment (moon0326) [reply plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/api/reports-categories.php:190]: Yes, it's causing other tests to fail. This isn't really needed for this test to pass.

File: plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/api/product-attributes.php (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/api/product-attributes.php]: Updated it in https://github.com/woocommerce/woocommerce/pull/32409/commits/150142c139d7ecded719fbd88749386559a064b3

File: plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/reports/class-wc-tests-reports-orders-stats.php (hunk 3 above)
Comment (moon0326) [reply plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/reports/class-wc-tests-reports-orders-stats.php]: Thank you for catching it! Yes, it's a typo. Updated in https://github.com/woocommerce/woocommerce/pull/32409/commits/8632950ab8a3d6a47d37801ca5aa8ee10330ef4f

**Comment by moon0326** [comment]
> Do we need to run other commands before running this?

Yes :) You need to set up the test environment. Try steps from https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/README.md

**Review by chihsuan — APPROVED** [review]
Thanks for fixing this! @moon0326  I have also run this in my local. Looks good. 💯

**Pull request merged**